### PR TITLE
Add `ServerInfo` type

### PR DESCRIPTION
**Pull request by atburke**
This PR adds a new `ServerInfo` type as described in [RFD 105](https://github.com/gravitational/teleport/blob/master/rfd/0105-ec2-tags.md). In later PRs, `ServerInfo`s will be matched to discovered servers and used to overwrite their labels.

### CODE REVIEW

**Comment by atburke** [comment]
Friendly ping @kimlisa @smallinsky @strideynet @tobiaszheller

**Review by smallinsky — APPROVED** [review]

File: lib/services/local/presence.go [hunk 1]
```diff
@@ -138,6 +139,112 @@ func (s *PresenceService) DeleteNamespace(namespace string) error {
 	return trace.Wrap(err)
 }
 
+// GetServerInfos returns a stream of ServerInfos.
+func (s *PresenceService) GetServerInfos(ctx context.Context, pageSize int) stream.Stream[types.ServerInfo] {
```
Comment (espadolini) [inline lib/services/local/presence.go:143]: Why is `pageSize` a parameter to this, if it's actually not going to change anything for the caller?

**Review by strideynet — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -5497,3 +5497,36 @@ message WatchStatusSpecV1 {
...
+  ServerInfoSpecV1 Spec = 5 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "spec"
+  ];
+}
+
+// ServerInfoSpecV1 contains fields used to match Nodes to this ServerInfo.
+message ServerInfoSpecV1 {
+  // AWSInfo contains attributes to match to an EC2 instance.
+  message AWSInfo {
```
Comment (strideynet) [inline api/proto/teleport/legacy/types/types.proto:5524]: nit: As this is within the context of `ServerInfoSpecV1`, I'd be tempted to just name this AWS for brevity. But this is a very minor nit.

File: lib/services/local/presence.go [hunk 3]
```diff
@@ -138,6 +139,85 @@ func (s *PresenceService) DeleteNamespace(namespace string) error {
 	return trace.Wrap(err)
 }
 
+// GetServerInfos returns a stream of ServerInfos.
+func (s *PresenceService) GetServerInfos(ctx context.Context) stream.Stream[types.ServerInfo] {
+	const pageSize = 100
```
Comment (espadolini) [inline lib/services/local/presence.go:144]: I think we can do 1000 without too much strain.

```suggestion
	const pageSize = apidefaults.DefaultChunkSize
```

**Review by espadolini — APPROVED** [review]

File: lib/services/local/presence.go (hunk 3 above)
Comment (strideynet) [reply lib/services/local/presence.go:144]: Agreed - and keeps this consistent with other methods.

File: api/proto/teleport/legacy/types/types.proto (hunk 2 above)
Comment (atburke) [reply api/proto/teleport/legacy/types/types.proto:5524]: I'd prefer that too, but protobuf doesn't allow nested types and fields to have the same name.

Comment (strideynet) [reply api/proto/teleport/legacy/types/types.proto:5524]: Oops, I forgot about that. I think I ran into similar on the join token stuff :(

**Pull request merged**